app/connector/mongodb/src/main/java/io/syndesis/connector/mongo/meta/MongoDBMetadataRetrieval.java

```diff
@@ -63,33 +69,70 @@ protected SyndesisMetadata adapt(CamelContext context, String componentId, Strin
         return meta.orElse(defaultMeta);
     }
 
-    private static SyndesisMetadata buildDatashape(String actionId, DataShape schema) {
-        if (actionId.endsWith("find")) {
-            return SyndesisMetadata.of(MongoDBMetadataRetrieval.criteria(), schema);
-        } else if (actionId.endsWith("insert")) {
+    public SyndesisMetadata buildDatashape(String actionId, DataShape schema, Map<String, Object> properties) {
+        if (isFilterAction(actionId)) {
+            return SyndesisMetadata.of(
+                MongoDBMetadataRetrieval.criteria(ConnectorOptions.extractOption(properties,"filter")),
+                schema);
+        } else if ("io.syndesis.connector:connector-mongodb-insert".equals(actionId)) {
             return SyndesisMetadata.of(schema);
         } else if (actionId.contains("consumer")) {
-            return SyndesisMetadata.outOnly(schema);
+            return SyndesisMetadata.of(none("N/A"), schema);
         } else {
             throw new IllegalArgumentException(String.format("Could not find any dynamic metadata adaptation for action %s", actionId));
         }
     }
 
-    private static DataShape any(String name) {
+    @SuppressWarnings("MethodCanBeStatic")
```

Comment:
Why not make the method static?

Reply:
I feel that is semantically an extension of `buildDatashape` method, and it's there to have a more concise coding.

Reply:
I don't understand what that means can you elaborate? I'm proposing that instead of `isFilterAction` being a `private` method if we can make it `private static` method instead of adding a `@SuppressWarnings("MethodCanBeStatic")`.

Reply:
I meant to say that it's a method used only [here](https://github.com/syndesisio/syndesis/blob/e4485cb2b206002ad9da24a7e6541d1395a2961a/app/connector/mongodb/src/main/java/io/syndesis/connector/mongo/meta/MongoDBMetadataRetrieval.java#L72-L73). The way I see this is that the `isFilterAction` is semantically part of the `MongoDBMetadataRetrieval` instance as I could have inlined the code into the `buildDatashape`. I do expect in fact that compiler would inline the source code at compilation time as that was not intended to be static on purpose.

Reply:
I'm sorry I've never come across the use of the word _semantically_ in context of Java methods, so I really don't understand. What do you mean that the method that is not static is semantically a part of a class vs a method that is static not being semantically a part of a class?

This method is a private method and it doesn't access any instance state, I don't see a valid reason for not making it static vs increasing the amount of technical debt by adding the SuppressWarnings annotation.

Reply:
The PR was merged automatically by the bot already.

But, just to clarify, I was meaning that the [final goal would be the this](https://github.com/squakez/syndesis/compare/feature/ENTESB-11589_filters...squakez:experiment/inline_method) - and I'd say that the compiler would take care on its own to do that. I can see too many static methods around just because PMD is forcing to go in that direction making our development too procedural. The design I chose was different, and the only way to achieve that was adding a `SuppressWarnings` annotation.

If you still feel that it must considered as technical debt, please log an issue and we will review that as the PR is already merged unfortunately.